lib/worker/master.rb

```diff
@@ -0,0 +1,90 @@
+# frozen_string_literal: true
+module Worker
+  # The Master Worker is the entrypoint to trade capture. The single 'work!' method will block continuously.
+  class Master
+    include ::ErrorNotifierMethods
+    include ::Logging
+
+    def initialize
+      Logging.logger.info { 'Creating Worker::Master' }
+      @paused = false
+      @active_accounts = []
+    end
+
+    # work is a blocking method. The 'fetch' and 'pause' methods have custom timeouts and are implemented
+    # in each method.
+    def work!
+      loop do
+        log_activity
+        inside_operating_window? ? fetch : pause
+      end
+    end
+
+    private
+
+    def fetch
+      fetch_active_accounts!.each do |account_hash|
+        fetch_trades_for_account!(account_hash)
+      end
+      sleep_before_next_trade_capture
```

Comment:
Worst case this will be {time to fetch all accounts} + {sleep time} for an account to come around again. It may make sense to time the above work and then subtract that from the default sleep time with min 0 sleep.